src/compiler/checker.ts

```diff
@@ -32601,7 +32604,7 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
         const isTaggedTemplate = node.kind === SyntaxKind.TaggedTemplateExpression;
         const isDecorator = node.kind === SyntaxKind.Decorator;
         const isJsxOpeningOrSelfClosingElement = isJsxOpeningLikeElement(node);
-        const reportErrors = !candidatesOutArray;
+        const reportErrors = !isInferencePartiallyBlocked && !candidatesOutArray;

```

Comment:
alternatively, we could check `checkMode` but `ContextFlags.Completions` isn't currently passed down as any kind of `checkMode` (whereas `CheckMode.IsForStringLiteralArgumentCompletions` already exists). So this would require some additional plumbing and I'm not sure if that's desired